### ISSUE TITLE
Wrong ternary suggested

### ISSUE THREAD

**Issue by sami-evangelista** [PyCQA/pylint#5200]
### Bug description

```python
# pylint: disable=missing-module-docstring,invalid-name

a = True
b = False
c = True
x = (a and b) or c
print(x)  #  x == True
print(b if a else c)  #  ternary suggested == False
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:6:0: R1706: Consider using ternary (b if a else c) (consider-using-ternary)
```

### Expected behavior

Success.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.9.7 (default, Aug 30 2021, 00:00:00) 
[GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]
```

### OS / Environment

Fedora 34
